boomerang/src/state_machine/tests.rs

```diff
@@ -0,0 +1,73 @@
+use crate::state_machine::{
+    BoomerangSetup, DEFAULT_PRINCIPAL_ID, ONE_MONTH_SECONDS, USER_PRINCIPAL_ID,
+};
+use crate::{E8S, TRANSFER_FEE};
+use ic_state_machine_tests::PrincipalId;
+use icp_ledger::AccountIdentifier;
+use icrc_ledger_types::icrc1::account::Account;
+
+#[test]
+fn check_e2e() {
+    let boomerang = BoomerangSetup::new();
+
+    let caller = PrincipalId::new_user_test_id(USER_PRINCIPAL_ID);
+    let minter = PrincipalId::new_user_test_id(DEFAULT_PRINCIPAL_ID);
+
+    // WaterNeuron initialization
+    assert!(boomerang
+        .icp_transfer(
+            minter.0,
+            None,
+            3 * E8S,
+            AccountIdentifier::new(boomerang.water_neuron_id.into(), None)
+        )
+        .is_ok());
+
+        boomerang.advance_time_and_tick(60*60);
+
+    let account_id = boomerang.get_staking_account_id(caller.0);
+
+    assert!(boomerang
+        .icp_transfer(caller.0, None, 1_000 * E8S, account_id)
+        .is_ok());
+
+    assert!(boomerang.notify_icp_deposit(caller.0).is_ok());
+
+    boomerang.advance_time_and_tick(60*60);
```

Comment:
this is probably not formatted

Reply:
some parts of the code were not but here it was. I formatted everything now.